tools/nl/embeddings/utils.py

```diff
@@ -172,4 +173,70 @@ def download_model_from_gcs(ctx: Context, model_folder_name: str) -> str:
       continue
     blob.download_to_filename(os.path.join(directory, file_split[-1]))
 
-  return os.path.join(local_dir, model_folder_name)
\ No newline at end of file
+  return os.path.join(local_dir, model_folder_name)
+
+
+def build_embeddings(ctx, texts: List[str], dcids: List[str]) -> pd.DataFrame:
+  assert len(texts) == len(dcids)
+
+  embeddings = ctx.model.encode(texts, show_progress_bar=True)
+  embeddings = pd.DataFrame(embeddings)
+  embeddings[DCID_COL] = dcids
+  embeddings[COL_ALTERNATIVES] = texts
+  return embeddings
+
+
+def get_or_download_model_from_gcs(ctx: Context, model_version: str) -> str:
+  """Returns the local model path, downloading it if needed.
+  
+  If the model is already downloaded, it returns the model path.
+  Otherwise, it downloads the model to the local file system and returns that path.
+  """
+  tuned_model_path: str = ""
+
+  # Check if this model is already downloaded locally.
+  if os.path.exists(os.path.join(ctx.tmp, model_version)):
+    tuned_model_path = os.path.join(ctx.tmp, model_version)
+    print(f"Model already downloaded at path: {tuned_model_path}")
+  else:
+    print("Model not previously downloaded locally. Downloading from GCS.")
+    tuned_model_path = download_model_from_gcs(ctx, model_version)
```

Comment:
is this a local function now?  if so, have `_` prefix?

Reply:
Done.